File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_BuffEffect.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_BuffEffect implements ITableOperate {
 public final static String TAB_FILE_DATA = "/buff_effect.txt";
 public static final int ID_BUFF_TYPE = 0 ;
public static final int ID_EFFECT_TYPE = 1 ;
public static final int ID_EFFECT_POS = 2 ;
public static final int ID_BUFF_ANIM = 3 ;
public static final int ID_BUFF_LIB = 4 ;
public static final int MAX_RECORD = 5 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private String m_BuffAnim;
 public String GetBuffAnim() { return m_BuffAnim;}

private String m_BuffLib;
 public String GetBuffLib() { return m_BuffLib;}

private int m_BuffType;
 public int GetBuffType() { return m_BuffType;}

private int m_EffectPos;
 public int GetEffectPos() { return m_EffectPos;}

private int m_EffectType;
 public int GetEffectType() { return m_EffectType;}

public boolean LoadTable(HashMap<Integer, Table_BuffEffect> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_BuffEffect> _hash = (HashMap<Integer, Table_BuffEffect> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_BuffEffect _values = new Table_BuffEffect();
 _values.m_BuffAnim =  (String)valuesList.get(ID_BUFF_ANIM);
_values.m_BuffLib =  (String)valuesList.get(ID_BUFF_LIB);
_values.m_BuffType =  Integer.parseInt((String)valuesList.get(ID_BUFF_TYPE));
_values.m_EffectPos =  Integer.parseInt((String)valuesList.get(ID_EFFECT_POS));
_values.m_EffectType =  Integer.parseInt((String)valuesList.get(ID_EFFECT_TYPE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/tabletools/CodeTable/CSharp/Table_BuffEffect.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_BuffEffect : ITableOperate
{ private const string TAB_FILE_DATA = "buff_effect.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_BUFF_TYPE,
ID_EFFECT_TYPE,
ID_EFFECT_POS,
ID_BUFF_ANIM,
ID_BUFF_LIB,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private string m_BuffAnim;
 public string BuffAnim { get{ return m_BuffAnim;}}

private string m_BuffLib;
 public string BuffLib { get{ return m_BuffLib;}}

private int m_BuffType;
 public int BuffType { get{ return m_BuffType;}}

private int m_EffectPos;
 public int EffectPos { get{ return m_EffectPos;}}

private int m_EffectType;
 public int EffectType { get{ return m_EffectType;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_BuffEffect _values = new Tab_BuffEffect();
 _values.m_BuffAnim =  valuesList[(int)_ID.ID_BUFF_ANIM] as string;
_values.m_BuffLib =  valuesList[(int)_ID.ID_BUFF_LIB] as string;
_values.m_BuffType =  Convert.ToInt32(valuesList[(int)_ID.ID_BUFF_TYPE] as string);
_values.m_EffectPos =  Convert.ToInt32(valuesList[(int)_ID.ID_EFFECT_POS] as string);
_values.m_EffectType =  Convert.ToInt32(valuesList[(int)_ID.ID_EFFECT_TYPE] as string);

 _hash.Add(nKey,_values); }


}
}

